tests/console/yast2_lan_device_settings.pm

```diff
@@ -46,12 +48,18 @@ sub run {
     send_key "alt-t";    # select static IP address option
     send_key "tab";
     type_string $static_ip;
+    send_key 'alt-o';
+    wait_screen_change(sub {
+            send_key 'backspace' for (0 .. length 'localhost') }, 2);
+    type_string $static_hostname;
     send_key "alt-n";    # next
     assert_screen 'static-ip-address-set';
     close_yast2_lan();
 
     # verify that static IP has been set
     assert_script_run "ip a | grep $static_ip";
+    # verify that hostname for static ip is recorded in /etc/hosts
+    $is_set_in_etc_host->($static_hostname) or die "Static hostname was not written to /etc/hosts file!\n";
```

Comment:
I would write `$static_hostname` in the message too. Actually you could call assert_script in `is_set_in_etc_host` subroutine and define `fail_message`

Reply:
My intention is to leave the caller to process the return code, since the anonymous sub should only let you know the state. And it is not used in homogeneous way, once it should die, `/etc/hosts` should be updated accordingly after yast2 lan finished its first iteration. And the second time is up to highlight the softfailure.

Reply:
Makes sense.